index.html

```diff
@@ -270,6 +263,60 @@ <h4>
             or bypassing the UI if there is only a single share target.
           </div>
         </section>
+        <section>
+          <h3>
+            Validate share data
+          </h3>
+          <p>
+            To <dfn>validate share data</dfn> with |data:ShareData| and
+            |base:URL|, run the following steps:
+          </p>
+          <ol class="algorithm">
+            <li>If none of |data|'s members {{ShareData/title}},
+            {{ShareData/text}}, or {{ShareData/url}} or {{ShareData/files}} are
+            present, return false.
+            </li>
+            <li>Let |titleTextOrUrl:boolean| be true if any of
+            {{ShareData/title}}, or {{ShareData/text}}, or {{ShareData/url}} is
+            present.
+            </li>
+            <li>If |data|'s {{ShareData/files}} member is present:
+              <ol>
+                <li>If |titleTextOrUrl| is false and |data|'s
+                {{ShareData/files}} member is empty, return false.
```

Comment:
This line is new, right? Maybe better to split it out as a new PR for easier tracking?

Reply:
I think this should stay. The spec already deals with files, so it's appropriate to check this here.

Reply:
I mean, `If |titleTextOrUrl| is false` part is new and thus does changes behavior.

Reply:
Ah, no, this is just a refactor of step 6 and 7 of the current spec:

6 .If none of data's members title, text, or url or files are present, return a promise rejected with a TypeError.
7. If data's files member is present:
7.1 If data's files member is empty, or if the implementation does not support file sharing, return a promise rejected with a TypeError, and abort these steps.

Reply:
The current spec says `{ text: "foo", files: [] }` is invalid, and this patch makes it valid by checking other field values.

Reply:
Ah right, still. Respectfully, I'd like to keep this as is.